python_modules/dagster/dagster_tests/core_tests/resource_tests/pythonic_resources/test_binding.py

```diff
@@ -585,3 +585,28 @@ def my_asset(context, my_resource: MyResourceWithDefault):
     )
 
     assert executed["yes"]
+
+
+class MyModuleLevelResource(ConfigurableResource):
+    str_field: str
+
+
+# Note that an explicit string annotation has the same effect as defining a resource in a module
+# using `from __future__ import annotations`. This test will only work against a module-scoped
+# resource-- this is a hard limitation of string annotations in Python as of 2023-07-06 and Python
+# 3.11.
```

Comment:
what does the error look like when you use `from __future__ import annotations` with non module scope type? or i guess just a typo-ed string ?

Reply:
I tried to explain the situation in detail here: https://github.com/dagster-io/dagster/pull/14675

But here is an excerpt (you can mentally sub out the quoted annotations `"Foo"` and `"Bar"` with `from __future__ import annotations`-- they have the same effect):

```
from typing import get_type_hints

class Foo:
    ...

def foo(x: "Foo"):
     ...

# works to dereference "Foo" since referent `Foo` is in module-level scope
get_type_hints(foo)  # => {'x': <class '__main__.Foo'>}

def make_bar():
    class Bar:
        ...

    def bar(x: "Bar"):
        ...
    return bar

bar = make_bar()

# does not work to dereference "Bar" since referent `Bar` vanished with local scope
get_type_hints(bar)  # => NameError: name 'Bar' is not defined.
```

Reply:
I was just asking to figure out what the end user experience would be for someone hitting this corner